cli/crates/cli/src/introspect.rs

```diff
@@ -28,11 +28,52 @@ fn introspect_local() -> Result<(), CliError> {
     Ok(())
 }
 
-fn introspect_remote(url: &str, headers: &[(&str, &str)]) -> Result<(), CliError> {
+fn introspect_remote(url: &str, headers: &[(&str, &str)], no_color: bool) -> Result<(), CliError> {
     let operation = graphql_introspection::introspect(url, headers);
 
     match Runtime::new().unwrap().block_on(operation) {
-        Ok(result) => Ok(println!("{result}")),
+        Ok(result) => {
+            print_introspected_schema(&result, no_color);
+            Ok(())
+        }
         Err(e) => Err(CliError::Introspection(e)),
     }
 }
+
+fn print_introspected_schema(sdl: &str, no_color: bool) {
+    use syntect::util::{as_24_bit_terminal_escaped, LinesWithEndings};
+
+    // No highlighting when stdout is not a tty (likely a pipe) or when explicitly requested.
+    if no_color || !atty::is(atty::Stream::Stdout) || no_color_env_var() {
+        println!("{sdl}");
+        return;
+    }
+
+    const GRAMMAR: &str = include_str!("introspect/graphql_grammar.yml");
+
+    let mut builder = syntect::parsing::SyntaxSetBuilder::new();
+    builder.add(syntect::parsing::SyntaxDefinition::load_from_str(GRAMMAR, false, Some("graphql")).unwrap());
+    let syntax_set = builder.build();
+
+    let graphql = syntax_set
+        .find_syntax_by_extension("graphql")
+        .expect("graphql syntax to be bundled");
+    let theme_set = syntect::highlighting::ThemeSet::load_defaults();
+    let theme = &theme_set.themes["Solarized (dark)"];
+
+    let mut highlighter = syntect::easy::HighlightLines::new(graphql, theme);
+
+    for line in LinesWithEndings::from(sdl) {
+        let ranges = highlighter
+            .highlight_line(line, &syntax_set)
+            .expect("line to be highlightable");
+
+        let escaped = as_24_bit_terminal_escaped(&ranges[..], false);
+        print!("{}", escaped);
+    }
+}
+
+/// https://no-color.org/
+fn no_color_env_var() -> bool {
+    std::env::var("NO_COLOR").is_ok_and(|s| !s.is_empty())
```

Comment:
Would it be possible to use a longer var name here?

Reply:
changed to `value`.